eng/references.targets

```diff
@@ -29,6 +29,21 @@
                       Condition="'$(IsSourceProject)' == 'true' and '@(_coreLibProjectReference)' != ''" />
   </ItemGroup>
 
+  <ItemGroup Condition="'@(PackageReference)' != ''">
+    <!-- Whenever a package with a transitive reference to System.Runtime.CompilerServices.Unsafe is referenced and there is no
+         direct package reference to System.Runtime.CompilerServices.Unsafe, reference the Unsafe package so that libraries build
+         against the latest available version (6.0.* instead of the old transitive reference which has a 4.5.* version). -->
+    <PackageReference Include="System.Runtime.CompilerServices.Unsafe"
```

Comment:
Do we really need this? Shouldn't the package have a real PackageReference on Unsafe to the latest version? That way customers who use that library get the latest version?

Basically, I'm wondering why we need this in a central location AND in all the projects you are changing.

Reply:
> Basically, I'm wondering why we need this in a central location AND in all the projects you are changing.

This statement affects projects that don't declare a direct dependency on the Unsafe package. Basically all the ProjectReferences that were removed in https://github.com/dotnet/runtime/pull/64861/files?file-filters%5B%5D=.csproj&show-deleted-files=true&show-viewed-files=true which had the following comment: `<!-- S.R.C.Unsafe isn't a primary but transitive dependency and this P2P makes sure that the live version is used. -->`.

In main, these projects currently consume the Unsafe 4.5.3 package transitively instead of the latest 6.0.0 one when i.e. System.Memory which depends on Unsafe is referenced.

Reply:
But why is the PackageReference `PrivateAssets="all"` then? Shouldn't the package actually get a real PackageReference to the latest? So the customers get the latest?  It won't work if we compile with the latest, but the old version is restored.